src/server/components/dashboard/listsItems/controllers.ts

```diff
@@ -113,14 +114,24 @@ export async function listItemGetController(req: Request, res: ListItemRes): Pro
     list,
     listItem,
     isPinned,
-    actionButtons: actionButtonsForStatus ?? [],
+    actionButtons: addArchiveOption(listItem, actionButtonsForStatus) ?? [],
     requestedChanges,
     error,
     title: serviceTypeDetailsHeading[listItem.type] ?? "Provider",
-    details: getDetailsViewModel(listItem),
+  details: getDetailsViewModel(listItem),
     csrfToken: getCSRFToken(req),
   });
 }
+
+function addArchiveOption(listItem: ListItemWithHistory, actionButtonsForStatus: string[]): string[] {
```

Comment:
Was there a reason why this was added as a function? it looks like it can just be added to `actionButtons`.